src/components/menu/menu.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { ReactComponentElement, ReactNode, useState } from "react"
+import { MdMenu, MdHome, MdPeople, MdMail } from "react-icons/md"
+
+import styles from "./menu.module.scss"
+
+type MenuItem = {
+  name: string
+  icon: ReactComponentElement<any>
+  path: string
+}
+
+export const Menu: React.FC = () => {
+  const [ collapsed, setCollapsed ] = useState<boolean>(true)
+  const [ toggleDisabled, setToggleDisabled ] = useState<boolean>(false)
+
+  const menuItems: MenuItem[] = [
+    {
+      name: 'O nas',
+      icon: <MdPeople />,
+      path: '#1',
+    },
+    {
+      name: 'Główna',
+      icon: <MdHome />,
+      path: '#2',
+    },
+    {
+      name: 'Kontakt',
+      icon: <MdMail />,
+      path: '#3',
+    },
+  ]
+
+  const generateItems = (items: MenuItem[]): ReactNode => {
+    return items.map(item => {
+      return (
+        <li className={styles.menu__item} key={item.path}>
+          <a href={item.path} className={styles.menu__link}>
+            <span className={styles.menu__link__icon}>{ item.icon }</span>
+            <span className={styles.menu__link__text}> { item.name } </span>
+          </a>
+        </li>
+      )
+    })
+  }
+
+  const toggleMenu = () => {
+    if (toggleDisabled) return
+
+    setToggleDisabled(true)
+    setCollapsed(!collapsed)
+
+    setTimeout(() => {
+      setToggleDisabled(false)
+    }, 300)
+  }
+
+  return (
+    <>
```

Comment:
what does that do ?

Reply:
React component, when you define html template, requires single main element. 
Earlier almost always we use a `div` element to wrap multiple nested tags. But this generate lots of unnecessary nesting in rendered html file.

Thats why they implemented `<>` - which does not generate anything at all :D 

Lets say this is something like `<ng-container>` in angular xD

Reply:
can we close this one please?

Reply:
Hard to say